pkg/kubelet/kubelet.go

```diff
@@ -2921,33 +2924,43 @@ func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, v1.PodResizeStatus, string)
 func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontainer.PodStatus) (*v1.Pod, error) {
 	allocatedPod, updated := kl.allocationManager.UpdatePodFromAllocation(pod)
 
-	if !updated {
-		// Desired resources == allocated resources. Check whether a resize is in progress.
+	// Always check whether a resize is in progress so we can set the PodResizeInProgressCondition
+	// accordingly.
+	var err error
+	defer func() {
```

Comment:
closures on deferred functions can be confusing. I'd prefer to reference named return values here, in case code below returns without setting `err` or `allocatedPod` directly (I think that's the right way to capture it?)

Reply:
updated